actors/builtin/storage_market/multiset.go

```diff
@@ -0,0 +1,123 @@
+package storage_market
+
+import (
+	cid "github.com/ipfs/go-cid"
+	errors "github.com/pkg/errors"
+	cbg "github.com/whyrusleeping/cbor-gen"
+
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+)
+
+type Multiset struct {
```

Comment:
This isn't quite a multiset (which would be a map of keys to ints), but a Set-Multimap: a multimap with set semantics for the values. And specifically with integer values. This contrasts with the List-Multimap that the existing multimap uses. Please rename to SetMultimap (and feel free to rename the other to ListMultimap). (cf [Guava](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/Multimap.html))

For its use as a deal-by-client index, I think a HAMT may be more appropriate for the  value type: deal ids for a party will be sparse and large magnitude. You can use the existing `Set` ADT.